arangod/Agency/Agent.h

```diff
@@ -138,7 +138,7 @@ class Agent final : public arangodb::Thread, public AgentInterface {
   void removeTrxsOngoing(Slice trxs) noexcept;
 
   /// @brief Check whether a trx is ongoing.
-  bool isTrxOngoing(std::string& id);
+  bool isTrxOngoing(std::string const& id) const noexcept;
```

Comment:
can't we use a `std::string_view` instead?

Reply:
We could, but the function internally will convert the `id` to a string anyway, because it does a lookup for the id in an `std::unordered_set<std::string>`. So without heterogenous lookups, we won't save any conversion here.

Reply:
😭 C++20 is soo close...